### ISSUE TITLE
Looking for turtles embedded addon leaks memory

### ISSUE THREAD

**Issue by melg8** [slowtorta/turtlewow-bug-tracker#6377]
**Please, provide a clear description what the bug is:**
LFT addon which is embedded into twow client leaks memory even when not actively used for party search.

**Steps to reproduce the behavior with as much detail as possible:**
1. Install [pfDebug](https://github.com/shagu/pfDebug) addon from shagu for memory consumption visualisation
2. Run twow original client with pfDebug addon enabled.
3. Right-Click on the Frame to open the Analyzer
4. Press Scan button on alanyzer
5. After Scan become active, close window with X on it.
6. Play some game without using LFT in any way. Just kill some mobs in location.
7. I suppose effect will be more present it large online numbers on server, cause more events will happen.
8. Open up Analyzer again and observe results like so: ![Снимок экрана (17601)](https://github.com/slowtorta/turtlewow-bug-tracker/assets/16599015/731db62c-10ba-4141-9801-4be624d12e45)

## Expected behavior
1. Expected to have around zero leak value for such permanent addition to the game.
2. As a workaround if 1st variant is not possible for some reason, expected to have toggle switch as for other addons to permanently disable this addon for times i dont look for party. (Or please post how i can disable it right now, if silly me just didnt saw that option available).

## Why its important
I play on old pc in which any lags are more noticible than more modern pcs. When any of addons leaks (uses in unreasonable way alot of memory) it makes garbage collector of game to run more frequently removing memory which were allocated by addon and actually already not used. Each garbage collection for me represents like 1-4 seconds freeze of game, which can be quite annoying in pve/raiding situation and brutaly horrifying in pvp situations. Especially its concerning cause this happens no matter what. And i dont have way to disable this addon, while i dont need this functionallity at all.

**Comment by shagu** [comment]
Greetings,

The way LFT works works and by the amount of dynamic data and the amount of messages it has to process, the expected behaviour of having near to zero leak is simply impossible. Even in the best case, you will always see memory being used for the string comparisons it has to do on the chat messages.

But more important is, to have clear testing here. Because it's not clear if all of this is the fault of LFT. Since LFT is using functions in its OnEvent script that are often overwritten by other addons it can be possible that other addons do the bad stuff here. I would at least suggest to disable **every** addon beside LFT and pfDebug and check the results again, just to be sure that no other addon's hook does bad memory calls.

**Comment by melg8** [comment]
I agree that its not clear if was LFT fault. But it would be nice to be able disable FLT anyway to see others actors involved and not struggle with LFT, especially if i dont use it a time.

**Comment by melg8** [comment]
@Shagu although i was able to reproduce this in av setting, by reloading with only pfUI and pfDebug addons enabled:
![изображение](https://github.com/slowtorta/turtlewow-bug-tracker/assets/16599015/3b5b3bf8-1183-418d-8028-ad8ad572b358)

**Comment by shagu** [comment]
Does that only happen in AV? Maybe LFT parses messages in AV it shouldn't parse.

When I debug my addons, I would go with the following approach in your case:

Comparing the "Average Memory Consumption" (in the tooltip)
- AV (Usual Addons enabled)
- AV (Only pfDebug + LFT)
- Open World (Usual Addons enabled)
- Open World (Only pfDebug + LFT)

Maybe that helps to further diagnose the issue, to get an idea of where to look at.

PS: Another idea to disable LFT would be: `/run LFT:UnregisterAllEvents()`

**Comment by melg8** [comment]
@shagu seems it related to your position in party or raid, when i try to move myself around from party 1 to party 2 it leaks each move, also when person just joined my party it also showed up with inital 250 kb leak.
![изображение](https://github.com/slowtorta/turtlewow-bug-tracker/assets/16599015/b18eb028-0a07-4421-95c4-e8455a3b9c30)
No addons except pfDebug is enabled on character, all loaded from login screen.

**Comment by twowTurtReynolds** [comment]
We have been working on the built in addons in the client, LFT included, we'll take a look regarding performance here.

I would suggest to the developer that LFT's logic be disabled when in a raid, or a battleground, which will help save computing power where most needed.
